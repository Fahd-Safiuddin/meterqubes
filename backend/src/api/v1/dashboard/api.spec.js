const server = require('../../../index')
const agent = require('supertest').agent(server)
const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjp7ImlkIjoyLCJwdWJsaWNBZGRyZXNzIjoiMHhhOD' +
  'BmZjA4ZDA4ODA5YjI5YzczMzEzNWIxYjkzYTA3Nzk5ZTVkMTdmIn0sImlhdCI6MTU1NDI4NzA5NH0.u-tN-7YKjA3MzkTrQlDx5Bkd4' +
  '1lVRE9C8fs3_9e6KuM'

describe('Dashboard (API)', function () {
  afterAll(async () => {
    await server.close()
  })

  it('get trade dashboard by market id', async () => {
    expect.assertions(2)
    const response = await agent.get('/api/v1/dashboard/trade/11')
      .set('Authorization', 'Bearer ' + token)
      .catch(err => console.error(err))

    expect(response.status).toEqual(200)
    expect(response.body).toMatchSnapshot()
  })

  it('get landing dashboard. All markets', async () => {
    expect.assertions(2)
    const response = await agent.get('/api/v1/dashboard/landing')
      .set('Authorization', 'Bearer ' + token)
      .catch(err => console.error(err))

    expect(response.status).toEqual(200)
    expect(response.body).toMatchSnapshot()
  })
})
