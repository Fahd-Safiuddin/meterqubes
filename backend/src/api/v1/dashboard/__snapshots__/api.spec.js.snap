// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard (API) get landing dashboard. All markets 1`] = `
Array [
  Object {
    "changeRate": "0.00",
    "lastPrice": "0.00",
    "token": "KCS",
  },
  Object {
    "changeRate": "0.00",
    "lastPrice": "0.00",
    "token": "ZRX",
  },
  Object {
    "changeRate": "0.00",
    "lastPrice": "0.00",
    "token": "BNB",
  },
  Object {
    "changeRate": "0.00",
    "lastPrice": "0.00",
    "token": "OMG",
  },
  Object {
    "changeRate": "0.00",
    "lastPrice": "0.00",
    "token": "HOT",
  },
  Object {
    "changeRate": "0.00",
    "lastPrice": "0.00",
    "token": "ENJ",
  },
  Object {
    "changeRate": "0.00",
    "lastPrice": "0.00",
    "token": "MKR",
  },
  Object {
    "changeRate": "0.00",
    "lastPrice": "0.00",
    "token": "BAT",
  },
  Object {
    "changeRate": "0.00",
    "lastPrice": "0.00",
    "token": "LINK",
  },
  Object {
    "changeRate": "0.00",
    "lastPrice": "0.00",
    "token": "WTC",
  },
  Object {
    "changeRate": "1.71",
    "lastPrice": "0.356000000",
    "token": "EMPR",
  },
]
`;

exports[`Dashboard (API) get trade dashboard by market id 1`] = `
Object {
  "changeRate": "1.71",
  "firstPrice": "0.350000000",
  "highPrice": "0.500000000",
  "lastPrice": "0.356000000",
  "lowPrice": "0.350000000",
  "priceChange": "0.006000000",
}
`;
